examples/controls_example/public/react_controls/data_controls/initialize_data_control.tsx

```diff
@@ -0,0 +1,140 @@
+import { DataView, DATA_VIEW_SAVED_OBJECT_TYPE } from '@kbn/data-views-plugin/common';
+import { DataViewsPublicPluginStart } from '@kbn/data-views-plugin/public';
+import { Filter } from '@kbn/es-query';
+import { SerializedPanelState } from '@kbn/presentation-containers';
+import { StateComparators } from '@kbn/presentation-publishing';
+
+import { ControlGroupApi } from '../control_group/types';
+import { initializeDefaultControlApi } from '../initialize_default_control_api';
+import { ControlApiInitialization, ControlStateManager, DefaultControlState } from '../types';
+import { openDataControlEditor } from './open_data_control_editor';
+import { DataControlApi, DefaultDataControlState } from './types';
+
+export const initializeDataControl = <EditorState extends object = {}>(
+  controlId: string,
+  controlType: string,
+  state: DefaultDataControlState,
+  editorStateManager: ControlStateManager<EditorState>,
+  controlGroup: ControlGroupApi,
+  services: {
+    core: CoreStart;
+    dataViews: DataViewsPublicPluginStart;
+  }
+): {
+  dataControlApi: ControlApiInitialization<DataControlApi>;
+  dataControlComparators: StateComparators<DefaultDataControlState>;
+  dataControlStateManager: ControlStateManager<DefaultDataControlState>;
+  serializeDataControl: () => SerializedPanelState<DefaultControlState>;
+} => {
+  const {
+    defaultControlApi,
+    defaultControlComparators,
+    defaultControlStateManager,
+    serializeDefaultControl,
+  } = initializeDefaultControlApi(state);
+
+  const panelTitle = new BehaviorSubject<string | undefined>(state.title);
+  const defaultPanelTitle = new BehaviorSubject<string | undefined>(undefined);
+  const dataViewId = new BehaviorSubject<string>(state.dataViewId);
+  const fieldName = new BehaviorSubject<string>(state.fieldName);
+  const dataViews = new BehaviorSubject<DataView[] | undefined>(undefined);
+  const filters = new BehaviorSubject<Filter[] | undefined>(undefined);
+
+  const dataControlComparators: StateComparators<DefaultDataControlState> = {
+    ...defaultControlComparators,
+    title: [panelTitle, (value: string | undefined) => panelTitle.next(value)],
+    dataViewId: [dataViewId, (value: string) => dataViewId.next(value)],
+    fieldName: [fieldName, (value: string) => fieldName.next(value)],
+  };
+
+  const stateManager: ControlStateManager<DefaultDataControlState> = {
+    ...defaultControlStateManager,
+    dataViewId,
+    fieldName,
+    title: panelTitle,
+  };
+
+  /**
+   * Fetch the data view + field whenever the selected data view ID or field name changes; use the
+   * fetched field spec to set the default panel title, which is always equal to either the field
+   * name or the field's display name.
+   */
+  dataViewId
+    .pipe(
+      combineLatestWith(fieldName),
+      distinctUntilChanged(
+        ([oldId, oldField], [newId, newField]) => oldId === newId && oldField === newField
+      ),
+      mergeMap(async ([currentDataViewId, currentFieldName]) => {
```

Comment:
nit: why `mergeMap` here instead of `switchMap`?

Reply:
The difference between `mergeMap` and `switchMap` is still a little shakey to me no matter how much I read the documentation TBH 😆 I think you're right though... Switched to `switchMap` in e72127c7c6c407068636f427b5b5a5e3e3be42ab

Reply:
`mergeMap` is for when you want _all_ events to come through. `switchMap` is for when you want newer events to cancel older events.